### PR TITLE
chatbot is now working from a seperate container

### PR DESCRIPTION
**Pull request by nnur**
- The chatbot service is now in a separate docker container
- It contains an amqpServer that communicates the messages to an amqpClient in the app container
From there the message is passed on with the chatbot socket.

Ba-da-bing.

### CODE REVIEW

File: chatbot/amqpServer.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  return amqp.connect(BROKER_URL)
+    .catch((err)=> {
+      return rabbitConnect();
+    });
+}
+
+
+rabbitConnect().then((connection)=> {
+  return connection.createChannel().then((ch)=> {
+    console.log('connection established');
```
Comment (moskalyk) [inline chatbot/amqpServer.ts]: Do we still need console logs?

File: app/microservices/chat/chatHandler.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+var chatClient = require('./amqpClient.js');
+
+module.exports = {
+  dispatchToAmqpClient: function(socket) {
```
Comment (dharness) [inline app/microservices/chat/chatHandler.ts]: it's unclear what this function does, given this name.

File: app/server.ts [hunk 3]
```diff
@@ -5,6 +5,8 @@ var express = require('express');
 var bodyParser = require('body-parser');
 var cors = require('./middleware/cors.js');
 var request = require('superagent');
+var testrabbit = require('./testrabbit.js');
```
Comment (dharness) [inline app/server.ts]: is this file still called testrabbit? Can we rename?

File: app/server.ts [hunk 4]
```diff
@@ -40,9 +42,19 @@ app.get('/script/:businessId', (req, res)=> {
 app.use('/message', messageController);
 app.use('/conversation', conversationController)
 
-app.get('/', (req, res)=> {
+app.get('/', (req, res) => {
   res.send('Howdie ho!!')
-})
+});
+
+app.get('/chat/:message', (req, res)=> {
```
Comment (dharness) [inline app/server.ts]: what is this route for?

File: app/services/chatService.ts [hunk 5]
```diff
@@ -63,7 +64,8 @@ module.exports = class ChatService {
             { status: 'online' });
         }
         else { 
-          chatBot.chatWith(socket); 
+          // chatBot.chatWith(socket);
```
Comment (dharness) [inline app/services/chatService.ts]: Please remove.

File: chatbot/chatEngine.ts [hunk 6]
```diff
@@ -0,0 +1,153 @@
+"use strict";
```
Comment (dharness) [inline chatbot/chatEngine.ts]: It appears that this file is a duplicate of the one above?

File: docker-compose.override.yml [hunk 7]
```diff
@@ -14,4 +14,10 @@ services:
   pg:
     build: ./postgres
 
+  chatbot:
```
Comment (dharness) [inline docker-compose.override.yml]: dont forget to rename this to your new service name

File: docker-compose.yml [hunk 8]
```diff
@@ -22,6 +22,8 @@ services:
   chindow:
     image: bablot/chindow
 
+  chatbot:
```
Comment (dharness) [inline docker-compose.yml]: same as above

File: chatbot/amqpServer.ts (hunk 1 above)
Comment (nnur) [inline chatbot/amqpServer.ts]: done

File: app/microservices/chat/chatHandler.ts (hunk 2 above)
Comment (nnur) [inline app/microservices/chat/chatHandler.ts]: done

File: npm-debug.log [hunk 9]
```diff
@@ -0,0 +1,24 @@
+0 info it worked if it ends with ok
```
Comment (dharness) [inline npm-debug.log]: rm this file please

**Comment by dharness** [comment]
LGTM, good luck with the merge

**Pull request merged**